docs/assemblies/assembly-overview.adoc

```diff
@@ -15,6 +15,21 @@ It takes on the responsibility of receiving, filtering, and forwarding messages.
 
 An API provides a convenient means for implementing custom logic within the proxy.
 
+[id='assembly-overview-compatability{context}']
+== Compatibility
+
+[id='assembly-overview-compatability-api{context}']
+=== APIs
+Kroxylicious follows https://semver.org/#semantic-versioning-200[Semantic Versioning] rules. While we are still in the `0.y.z` phase we still take API compatability very seriously, we aim to provide at least two minor releases between deprecation and the removal of that deprecated item.
+
+We also consider our configuration model to be a "public API", and thus it will follow the same Semantic Versioning and deprecation rules.
```

Comment:
Define what you think the "configuration model" is. Does it include the Java API, or are we just promising that we aim to support existing config files which make use of deprecated synatx for the duration of the deprecation period?

Also, because we support plugins, this is not entirely our promise to make. We should make it clear that configuration compatibility also depends on plugin authors.

Reply:
I was just intending to promise that config files are considered an API, how we parse them is up to us. That said its not purely my call to make, do you think we should include the Java API there as well?

You make a good point about plugin config being outside of our control. I'll clarify that.

Reply:
> do you think we should include the Java API there as well?

No, absolutely not! The trouble is how to phrase what the guarantee is without it being very convoluted. This is painful in part because we don't currently have a first class schema. So you have to read the java, know Jackson well enough to know what YAML it would accept, and then we're saying that any YAML accepted by version X will still be accepted in version Y.

```suggestion
We also consider our configuration file syntax (but not the Java code implementing it) to be a "public API", and thus it will follow the same Semantic Versioning and deprecation rules.
```